cdap-docs/tools/doc-cdap-default.py

```diff
@@ -0,0 +1,434 @@
+  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+  License for the specific language governing permissions and limitations under
+  the License.
+  -->
+""" % datetime.now().year
+
+
+class Item:
+    @staticmethod
+    def encode(value):
+        import cgi
+        if value:
+            return cgi.escape(value).encode('ascii', 'xmlcharrefreplace')
+        else:
+            return ''
+
+    def __init__(self, name='', value='', description ='', final=None):
+        self.name = name
+        self.value = self.encode(value)
+        self.description = self.encode(description)
+        self.final = final
+    
+    def __str__(self):
+        if self.final != None:
+            return "%s:\n%s (%s)\n%s" % (self.name, self.value, self.final, self.description)
+        else:
+            return "%s:\n%s\n%s" % (self.name, self.value, self.description)
+
+    def _append(self, name, value):
+        if self.__dict__[name]:
+            self.__dict__[name] = "%s\n%s" % (self.__dict__[name], value)
+        else:
+            self.__dict__[name] = value
+    
+    def rst(self):
+        name = "``%s``" % self.name
+        if self.value.find(' ') != -1:
+            value = BLOCK_START + BLOCK_JOIN.join(self.value.split()) + '``'
+        elif not self.value:
+            value = ''
+        else:
+            value = "``%s``" % self.value
+        rst = "%s%s\n%s%s\n%s%s" % (NAME_START, name,
+                                    VALUE_START, value,
+                                    DESC_START, self.description)
+        return rst
+    
+    def set_attribute(self, name, value):
+        v1 = "%s" % self.encode(value)
+        if name == 'description' and v1 == 'None':
+            v = ''
+        elif name == 'description':
+            v = " ".join(v1.split())
+        elif name == 'final':
+            v = (v1.lower() == 'true')
+        else:
+            v = v1
+        self.__dict__[name] = v
+            
+    def get_attribute(self, name):
+        if name == 'rst':
+            return self._rst()
+        else:
+            return self.__dict__[name]
+    
+    def append_value(self, value):
+        self._append("value", value)
+
+    def append_description(self, description):
+        self._append("description", description)
+    
+    def display(self):
+        print self.name
+
+
+class Section:
+
+    def __init__(self, name=''):
+        self.name = name.strip()
+        self.final = None
+
+    def rst(self):
+        CONFIG = ' Configuration'
+        name = self.name.strip()
+        if name.endswith(CONFIG):
+            name = name[0:-len(CONFIG)]
+        underline = "-" * len(name)
+        rst = "\n%s\n%s\n%s" % (name, underline, RST_TABLE_HEADER)
+        return rst
+
+    def display(self):
+        print "\nSection %s" % self.name
+       
+
+class PIParser(ET.XMLTreeBuilder):
```

Comment:
It stands for "Processing Instructions Parser", and came from this site: http://effbot.org/zone/element-pi.htm
Perhaps appropriate comment explaining it?